### PR TITLE
Add cluster-mode design document

### LINKED ISSUE #11: Support HA servers in active-active manner
Currently Yorkie only provides a single server. Servers can always be down, we need to set up the HA servers. Unlike other servers, Yorkie will receive many edit operations, so we will configure it in an active-active manner.

If we handle the following tasks, It seems like we can achieve set up HA servers.

- [x] Introduce distributed lock with lease
- [ ] Introduce distributed PubSub

### PR DESCRIPTION
**Pull request by hackerwins**
**What this PR does / why we need it**:

Writing a design document for Cluster-Mode.

**Which issue(s) this PR fixes**:

Fixes #11

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: design/cluster-mode.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: cluster-mode
+target-version: 0.1.3
```
Comment (hackerwins) [inline design/cluster-mode.md:3]:
```suggestion
target-version: 0.1.4
```

File: design/cluster-mode.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+title: cluster-mode
+target-version: 0.1.4
```
Comment (hackerwins) [inline design/cluster-mode.md:3]:
```suggestion
target-version: 0.1.6
```

**Review by dc7303 — APPROVED** [review]
LGTM 👍

**Review by mojosoeun — APPROVED** [review]

**Review by ppeeou — APPROVED** [review]
LGTM

**Pull request merged**